packages/admin-next/dashboard/src/routes/products/product-media/components/edit-product-media-form/edit-product-media-form.tsx

```diff
@@ -85,20 +86,17 @@ export const EditProductMediaForm = ({ product }: ProductMediaViewProps) => {
     if (filesToUpload.length) {
       const files = filesToUpload.map((m) => m.file) as File[]
 
-      // TODO: Implement upload to Medusa
-      // const uploads = await client.admin.uploads
-      //   .create(files)
-      //   .then((res) => {
-      //     return res.uploads
-      //   })
-      //   .catch((_err) => {
-      //     // Show error message
-      //     return null
-      //   })
-
-      const uploads = files.map((file) => ({
-        url: URL.createObjectURL(file),
-      }))
+      const uploads = await sdk.admin.uploads
+        .create({ files })
+        .then((res) => {
+          return res.files
+        })
+        .catch(() => {
+          form.setError("media", {
+            type: "invalid_file",
+            message: t("products.media.failedToUpload"),
+          })
+        })
```

Comment:
```suggestion
      const { files: uploads } = await sdk.admin.uploads
        .create({ files })
        .catch(() => {
          form.setError("media", {
            type: "invalid_file",
            message: t("products.media.failedToUpload"),
          })
        })
```

Reply:
👍 I'll open another PR to add upload support when creating a product, and I'll refactor it then anyway